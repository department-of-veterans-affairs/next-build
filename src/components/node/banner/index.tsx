import { useRef, useEffect, useState } from 'react'
import { VaBanner } from '@department-of-veterans-affairs/component-library/dist/react-bindings'
import { isEmpty } from 'lodash'
import { recordEvent } from '@/utils/recordEvent'
import { NodeMetaInfo, NodeResourceType } from '@/types/node'

const BannerComponent = ({ node }): JSX.Element => {
  const [isClicked, setIsClicked] = useState(false)
  const analyticsRef = useRef(null)

  useEffect(() => {
    function handler(event) {
      if (analyticsRef.current?.contains(event.target)) {
        setIsClicked(true)
      }
    }
    window.addEventListener('click', handler)
    return () => window.removeEventListener('click', handler)
  }, [])

  if (isClicked) {
    recordEvent({
      eventCategory: 'Banner',
      eventAction: 'Click',
      eventLabel: node.title,
    })
  }

  if (isEmpty(node)) return

  return (
    <VaBanner
      id={node.id}
      role="va-banner"
      showClose={node.field_dismissible_option != 'perm'}
      headline={node.title}
      type={node.field_alert_type}
      visible={true}
      windowSession={node.field_dismissible_option == 'dismiss-session'}
    >
      <div
        ref={analyticsRef}
        dangerouslySetInnerHTML={{ __html: node?.body?.processed }}
      />
    </VaBanner>
  )
}
export default BannerComponent

export const Meta: NodeMetaInfo = {
  resource: NodeResourceType.Banner,
  component: BannerComponent,
  params: null,
  collection: true,
}
