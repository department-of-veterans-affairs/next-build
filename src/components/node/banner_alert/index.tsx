import { isEmpty } from 'lodash'
import { useRef, useEffect, useState } from 'react'
import { recordEvent } from '@/utils/recordEvent'
import { regionBaseURL, trackLinks } from '@/utils/helpers'
import { VaBanner } from '@department-of-veterans-affairs/component-library/dist/react-bindings'

const BannerAlert = ({ node }): JSX.Element => {
  const [isClicked, setIsClicked] = useState(false)
  const analyticsRef = useRef(null)

  useEffect(() => {
    function handler(event) {
      if (analyticsRef.current?.contains(event.target)) {
        setIsClicked(true)
      }
    }
    window.addEventListener('click', handler)
    return () => window.removeEventListener('click', handler)
  }, [])

  if (isEmpty(node)) return

  const hideOnSubpages = node.field_alert_inheritance_subpages
  const alertType =
    node.field_alert_type === 'information' ? 'info' : node.field_alert_type
  const region = '/' + regionBaseURL(node.path.alias)
  const lastArg = node.path.alias.substring(node.path.alias.lastIndexOf('/'))
  const bodyEventData = {
    event: 'nav-alert-box-link-click',
    'alert-box-status': alertType,
    'alert-box-headline': node.title,
    'alert-box-headline-level': '3',
    'alert-box-background-only': 'false',
    'alert-box-closeable': 'false',
  }
  let body = trackLinks(node.field_body.processed, bodyEventData)
  let outputStatus = false
  let statusUrl = ''

  if (isClicked) {
    recordEvent({
      eventCategory: 'Banner',
      eventAction: 'View',
      eventLabel: node.title,
    })
  }

  node.field_banner_alert_vamcs?.map((vamcs) => {
    if (region == vamcs.field_office.path.alias) {
      outputStatus = true
    }
    if (hideOnSubpages && lastArg != region && lastArg != '/operating-status') {
      outputStatus = false
    }
    statusUrl = vamcs.path.alias
  })

  if (node.field_alert_operating_status_cta && statusUrl.length) {
    const ctaEventData = {
      event: 'nav-warning-alert-box-content-link-click',
      alertBoxHeading: `${node.title}`,
    }

    body += `<p>
          <a href='${statusUrl}' onClick='recordEvent(${JSON.stringify(
      ctaEventData
    )})'>
            Get updates on affected services and facilities
          </a>
      </p>`
  }

  if (node.field_alert_find_facilities_cta) {
    body += `
      <p>
        <a href="/find-locations">Find other VA facilities near you</a>
      </p>`
  }

  return (
    <VaBanner
      id={node.id}
      role="va-banner"
      showClose={node.field_alert_dismissable}
      headline={node.title}
      type={node.field_alert_type}
      visible={outputStatus}
      windowSession={node.field_alert_dismissable == 'dismiss-session'}
    >
      <div ref={analyticsRef} dangerouslySetInnerHTML={{ __html: body }} />
    </VaBanner>
  )
}
export default BannerAlert