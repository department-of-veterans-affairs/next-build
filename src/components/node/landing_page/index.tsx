import Link from 'next/link'

// Output a landing_page node as a link to itself, plus its teaser text.
const BenefitsHubLink = ({ node }) => {
  if (node) {
    return (
      <>
        <p className="vads-u-margin--0">
          <strong>
            <Link href={node.path?.alias ? node.path?.alias : ' '} passHref>
              <a>{node.field_home_page_hub_label}</a>
            </Link>
          </strong>
        </p>
        <p className="vads-u-margin--0">{node.field_teaser_text}</p>
      </>
    )
  }
}

// This should not be the default; we will have a full display eventually.
// Only here to meet required default rules.
export default BenefitsHubLink
