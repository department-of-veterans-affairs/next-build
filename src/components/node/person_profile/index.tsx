import { v4 as uuidv4 } from 'uuid'
import Image from '@/components/image'
import { DEV_PATH } from '@/lib/constants'
import { StaffProfiles } from '@/components/paragraph/staff_profile'

export const PersonProfile = ({ node }): JSX.Element => {
  if (!node) return
  const {
    title,
    field_body: fieldBody,
    field_email_address: fieldEmailAddress,
    field_phone_number: fieldPhoneNumber,
    field_suffix: suffix,
    field_description: fieldDescription,
    field_complete_biography_create: fieldCompleteBiographyCreate,
    field_intro_text: fieldIntroText,
    field_media: fieldMedia,
    field_photo_allow_hires_download: fieldPhotoAllowHiresDownload,
    field_complete_biography: fieldCompleteBiography,
  } = node

  const thumbnail = DEV_PATH + fieldMedia?.thumbnail?.uri?.url
  const image = DEV_PATH + fieldMedia?.image?.uri?.url
  const meta = fieldMedia?.thumbnail?.resourceIdObjMeta

  return (
    <div key={uuidv4()}>
      <h3>Person Profile</h3>
      <article className="usa-content">
        <div className="usa-grid usa-grid-full vads-u-margin-bottom--2 vads-u-display--flex vads-u-flex-direction--column medium-screen:vads-u-flex-direction--row">
          {fieldMedia?.thumbnail && (
            <div className="vads-u-margin-bottom--2 medium-screen:vads-u-margin-bottom--0 vads-u-margin-right--3">
              <Image
                className="person-profile-detail-page-image"
                src={thumbnail}
                alt={meta?.alt}
                title={meta.title}
                width={meta?.width}
                height={meta?.height}
              />
            </div>
          )}
          <div className="vads-u-display--flex vads-u-flex-direction--column">
            <h1 className="vads-u-font-size--xl vads-u-margin-bottom--0p5">
              {title}
              {suffix && [suffix]}
            </h1>
            {fieldDescription && (
              <p className="vads-u-font-size--lg vads-u-margin-bottom--0p5">
                {fieldDescription}
              </p>
            )}
            {fieldEmailAddress && (
              <p className="vads-u-font-size--lg vads-u-margin-bottom--0p5">
                <span className="vads-u-font-weight--bold">Email: </span>
                <a target="blank" href="mailto:{{ fieldEmailAddress }}">
                  {fieldEmailAddress}
                </a>
              </p>
            )}
            {fieldPhoneNumber && (
              <p
                className="
                vads-u-font-weight--regular
                vads-u-margin--0
                vads-u-margin-bottom--0p5"
              >
                <span className="vads-u-font-weight--bold">Phone: </span>
                <a href="tel:{{ fieldPhoneNumber }}">{fieldPhoneNumber}</a>
              </p>
            )}
          </div>
        </div>
        {fieldCompleteBiographyCreate && (
          <div className="va-introtext vads-u-margin-bottom--2">
            <p className="va-introtext vads-u-margin-bottom--0">
              {fieldIntroText}
            </p>
            <div
              className="vads-u-margin-bottom--2"
              dangerouslySetInnerHTML={{ __html: fieldBody.processed }}
            />
          </div>
        )}
        {fieldPhotoAllowHiresDownload && (
          <div
            className="vads-u-align-content--flex-end va-c-margin-top--auto vads-u-margin-bottom--2"
            id="download-full-size-photo-link"
          >
            <i
              className="va-c-social-icon fas fa-download"
              aria-hidden="true"
            ></i>
            <a href={image} download>
              Download full size photo
            </a>
          </div>
        )}
        {fieldCompleteBiography && (
          <div
            className="vads-u-align-content--flex-end va-c-margin-top--auto vads-u-margin-bottom--2"
            id="download-full-size-photo-link"
          >
            <i
              className="va-c-social-icon fas fa-download"
              aria-hidden="true"
            ></i>
            <div className="vads-u-align-content--flex-end va-c-margin-top--auto vads-u-margin-bottom--2">
              <i className="va-c-social-icon fas fa-download"></i>
              <a href={fieldCompleteBiography.url} download>
                Download full bio (PDF)
              </a>
            </div>
          </div>
        )}
      </article>
      <hr />
    </div>
  )
}

export const StaffNewsProfile = ({ node }): JSX.Element => {
  if (!node) return

  const { id, title, field_description: fieldDescription } = node

  return (
    <div className="vads-u-font-size--sm vads-u-margin-bottom--2p5">
      <div className="authored-by-line vads-u-margin-bottom--0p5 vads-u-font-weight--bold">
        By {title}
        {fieldDescription ? `, ${fieldDescription}` : null}
      </div>
    </div>
  )
}

export const StaffProfile = ({ paragraph }) => {
  return (
    <div key={uuidv4()}>
      <h3>Staff Profile</h3>
      <StaffProfiles paragraph={paragraph} />
    </div>
  )
}
