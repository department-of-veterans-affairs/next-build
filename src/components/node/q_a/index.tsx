/**
 * ### Overview
 * Resource QA is a node that displays a question and answer section.
 *
 * Resources QA expects nodes of type {@link NodeQA}.
 *
 * ### View modes
 * Full page: {@link NodeQAFull}
 *
 * ### Examples
 * @see https://va.gov/resources/how-are-pension-benefits-and-disability-compensation-different
 *
 */

/** These types/packages will import into all node components. */
import { NodeQA, NodeResourceType, NodeMetaInfo } from '@/types/node'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

/** These component includes are specific to this component. */
import Container from '@/components/container'
import { BenefitsHubLinks } from '@/components/partials/benefitHubsLinks'
import { Paragraph } from '@/lib/delegators/Paragraph'

/**
 * These components expect NodeQA as their input.
 */
type NodeQaProps = {
  node: NodeQA
}

export const ResourceWrapper = ({
  title,
  fieldAnswer,
  buttons,
  tags,
  linkTeaser,
  benefits,
}) => {
  return (
    <div id="content" className="interior" data-template="node-q_a">
      <Container className="container">
        <main className="va-l-detail-page">
          <div className="usa-grid usa-grid-full">
            <div className="usa-width-three-fourths">
              <div className="usa-content">
                {/** Add Search Widget Here */}
                <article className="vads-u-padding-x--1 large-screen:vads-u-padding-x--0">
                  <h1>{title}</h1>
                  <div dangerouslySetInnerHTML={fieldAnswer} />
                  <ul className="vads-u-margin-top--3 vads-u-margin-bottom--3 usa-unstyled-list">
                    {buttons}
                  </ul>
                  {tags}
                  {/*Section title*/}
                  <h2 className="vads-u-margin-y--0 vads-u-font-size--h3">
                    Related information
                  </h2>
                  {/*Links*/}
                  <ul className="usa-unstyled-list">{linkTeaser}</ul>
                  {benefits}
                </article>
              </div>
            </div>
          </div>
        </main>
      </Container>
    </div>
  )
}

export const Resources = ({ node }: NodeQaProps) => {
  /** Type narrowing; if we've managed to end up here with the wrong data, return. */
  if (node?.type !== NodeResourceType.QuestionAnswer) return

  const resources = {
    benefits: <BenefitsHubLinks nodes={node?.field_related_benefit_hubs} />,
    tags: <Paragraph paragraph={node?.field_tags} />,
    teaser: node?.field_related_information.map((paragraphLinkTeaser) => (
      <Paragraph
        key={paragraphLinkTeaser?.id}
        paragraph={paragraphLinkTeaser}
        componentParams={[{ boldTitle: false }, { sectionHeader: '' }]}
      />
    )),
    buttons: node.field_buttons.map((paragraphButton) => (
      <Paragraph key={paragraphButton.id} paragraph={paragraphButton} />
    )),
  }

  return (
    <ResourceWrapper
      title={node?.title}
      tags={resources?.tags}
      benefits={resources?.benefits}
      buttons={resources?.buttons}
      fieldAnswer={{
        __html: node?.field_answer?.field_wysiwyg?.processed,
      }}
      linkTeaser={resources?.teaser}
    ></ResourceWrapper>
  )
}

/** All nodes end with NodeMetaInfo: the name of the resource, the name of the component, and the parameters necessary for calling the resource. */
const params = new DrupalJsonApiParams().addInclude([
  'field_buttons',
  'field_related_information',
  'field_answer',
  'field_tags, field_tags.field_audience_beneficiares',
  'field_related_benefit_hubs',
  'field_contact_information',
])

/** Export information necessary to identify the component and query it.
 * See {@link NodeMetaInfo}
 */
export const Meta: NodeMetaInfo = {
  resource: NodeResourceType.QuestionAnswer,
  component: Resources,
  params: params,
}
