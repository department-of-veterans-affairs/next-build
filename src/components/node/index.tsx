import { Meta as NewsStoryMeta } from '@/components/node/news_story'
import { NodeTypes } from '@/types/node'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

/** General NodeProps to pass nodes into node components. */
interface NodeProps {
  node: NodeTypes
  viewMode?: string
}

/** We expect each Node component to define and export information about itself to help with discoverability. */
export interface NodeMetaInfo {
  resource: string
  component: ({ node, viewMode, ...props }: NodeProps) => JSX.Element
  params: DrupalJsonApiParams
}

/** Collect all imported node meta information. */
const nodeMetaIn: NodeMetaInfo[] = [NewsStoryMeta]

/** This interface enforces that the Node meta information is indexable by type. */
interface NodeMetaOut {
  [resource: string]: {
    component: ({ node }: NodeProps) => JSX.Element
    params: DrupalJsonApiParams
  }
}

/** Converts the meta information into a form indexed by resource type. Very possibly overwrought. */
export const nodeMeta: NodeMetaOut = nodeMetaIn.reduce((acc, current) => {
  const { resource, component, params } = current
  acc[resource] = {
    component: component,
    params: params,
  }
  return acc
}, {})

/** Generalized component. Look up the component the data requires and use it to render. */
export function Node({ node, viewMode = 'full', ...props }: NodeProps) {
  if (!node) {
    return null
  }

  const Component = nodeMeta[node.type].component

  if (!Component) {
    return null
  }

  return <Component node={node} viewMode={viewMode} {...props} />
}
