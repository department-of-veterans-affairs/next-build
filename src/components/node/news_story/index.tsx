/** These types/packages will import into all node components. */
import { NodeMetaInfo } from '@/components/node'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

/** These component includes are specific to this component. */
import { MediaImage } from '@/components/media'
import { StaffNewsProfile } from '@/components/node/person_profile'
import { NodeNewsStory } from '@/types/node'

type NodeNewsStoryProps = {
  node: NodeNewsStory
  viewMode?: string
}

/** Full page news story. */
export const NewsStoryFull = ({ node }: NodeNewsStoryProps) => {
  /** Type narrowing; if we've managed to end up here with the wrong data, return. */
  if (node.type !== 'node--news_story') return
  const dateTime = new Date(node.created).toLocaleDateString('en-us', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  })
  return (
    <>
      <div id="content" className="interior">
        <main className="va-l-detail-page va-facility-page">
          <div className="usa-grid usa-grid-full">
            {/* nav here */}
            <div className="usa-width-three-fourths">
              <article className="usa-content">
                <h1>{node.title}</h1>
                <MediaImage media={node.field_media} imageStyle={'2_1_large'} />
                <div className="vads-u-font-size--sm vads-u-margin-bottom--2p5">
                  {node.field_image_caption}
                </div>
                <StaffNewsProfile node={node.field_author} />
                <div className="created-line vads-u-margin-bottom--2p5">
                  <time dateTime="{ dateTime }">{dateTime}</time>
                </div>

                {/* {% include "src/site/facilities/story_social_share.drupal.liquid" %} */}

                <div className="usa-grid usa-grid-full vads-u-margin-bottom--2">
                  <div className="va-introtext">
                    <p>{node.field_intro_text}</p>
                  </div>
                  <div
                    className="full-story"
                    dangerouslySetInnerHTML={{
                      __html: node.field_full_story.processed,
                    }}
                  ></div>
                </div>
                {/* {% if fieldListing.entity.entityUrl.path %}
                        <a onClick="recordEvent({ event: 'nav-secondary-button-click' });" className="vads-u-display--block vads-u-margin-bottom--7" href="{{ fieldListing.entity.entityUrl.path }}" id="news-stories-listing-link">See all stories</a>
                    {% endif %} */}
              </article>
            </div>
          </div>
        </main>
      </div>
    </>
  )
}

/** General News Story component. Allows choice of different display components by the caller. */
export const NewsStory = ({ node, viewMode, ...props }: NodeNewsStoryProps) => {
  switch (viewMode) {
    case 'full':
      return <NewsStoryFull node={node} {...props} />
      break

    default:
      return null
  }
}

/** All nodes end with NodeMetaInfo: the name of the resource, the name of the component, and the parameters necessary for calling the resource. */
const params = new DrupalJsonApiParams().addInclude([
  'field_media',
  'field_media.image',
  'field_author',
])
export const Meta: NodeMetaInfo = {
  resource: 'node--news_story',
  component: NewsStory,
  params: params,
}
