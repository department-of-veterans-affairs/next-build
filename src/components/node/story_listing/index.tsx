/**
 * ### Overview
 * Story Listing represents an individual story within a Facility. These are used for human-interest articles.
 *
 * Story Listing expects nodes of type {@link NodeStoryListing}.
 *
 * ### View modes
 * Teaser: {@link StoryListing}
 *
 * ### Examples
 * @see https://va.gov/pittsburgh-health-care/stories/
 *


/** These types/packages will import into all node components. */
import { NodeMetaInfo, NodeResourceType } from '@/types/node'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'

/**
 * These components expect NodeStoryListing as their input.
 */

import Container from '@/components/container'
import { NewsStory } from '@/components/node/news_story'
import { isValidData } from '@/utils/helpers'

/** General Story Listing component. Allows choice of different display components by the caller. */
const StoryListing = ({ node, additionalNode }): JSX.Element => {
  if (!isValidData(node || additionalNode)) {
    return
  }

  return (
    <>
      {node.map((storyListing) => (
        <div key={storyListing.id} className="usa-grid usa-grid-full">
          {/*<div className="usa-width-three-fourths">*/}
          <h1>{storyListing.title}</h1>
          <div className="vads-l-grid-container--full">
            <div className="va-introtext">
              {storyListing.field_intro_text && (
                <p className="events-show">{storyListing.field_intro_text}</p>
              )}
              {additionalNode.length === 0 ||
                (additionalNode.length === null && (
                  <div className="clearfix-text">No stories at this time.</div>
                ))}
              <Container className="container">
                <ul className="usa-unstyled-list">
                  {additionalNode.map((newsStoryTeaser) => (
                    <NewsStory
                      node={newsStoryTeaser}
                      key={newsStoryTeaser.id}
                      viewMode="teaser"
                    />
                  ))}
                </ul>
              </Container>
            </div>
          </div>
          {/*</div>*/}
        </div>
      ))}
    </>
  )
}

export default StoryListing

/** All nodes end with NodeMetaInfo: the name of the resource, the name of the component, and the parameters necessary for calling the resource. */
const params = new DrupalJsonApiParams()
  .addFilter('status', '1')
  .addSort('created', 'DESC')

const newsStory = new DrupalJsonApiParams()
  .addInclude(['field_media', 'field_media.image', 'field_listing'])
  .addPageLimit(10)

/** Export information necessary to identify the component and query it.
 * See {@link NodeMetaInfo}
 */

export const Meta: NodeMetaInfo = {
  resource: NodeResourceType.StoryListing,
  component: StoryListing,
  params: params,
  additionalNode: NodeResourceType.NewsStory,
  additionalParams: newsStory,
  collection: true,
}
