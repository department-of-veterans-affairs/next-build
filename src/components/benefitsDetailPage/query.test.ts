/**
 * @jest-environment node
 */

import { BenefitsDetailPage } from '@/types/drupal/node'
import { queries } from '@/lib/drupal/queries'
import mockData from './mock.json'

const BenefitsDetailPageMock: BenefitsDetailPage = mockData

// remove if this component does not have a data fetch
describe('DrupalJsonApiParams configuration', () => {
  test('params function sets the correct include fields', () => {
    // TODO
  })
})

describe('BenefitsDetailPage formatData', () => {
  test('outputs formatted data', () => {
    expect(
      queries.formatData('node--benefits_detail_page', BenefitsDetailPageMock)
    ).toMatchSnapshot()
  })

  test('handles no answers correctly', () => {
    // TODO
  })
})
