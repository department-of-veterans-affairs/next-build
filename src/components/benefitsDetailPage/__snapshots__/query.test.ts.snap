// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BenefitsDetailPage formatter outputs formatted data with correct structure 1`] = `
{
  "administration": {
    "entityId": 1,
    "name": "Veterans Benefits Administration",
  },
  "alert": null,
  "breadcrumbs": [
    {
      "options": [],
      "title": "Home",
      "uri": "http://va-gov-cms.ddev.site/",
    },
    {
      "options": [],
      "title": "Check Your VA Claim or Appeal Status",
      "uri": "http://va-gov-cms.ddev.site/check-your-va-claim-or-appeal-status",
    },
  ],
  "contentBlock": [
    {
      "html": "<h2 id="how-to-check-your-va-claim-or-appeal-status-online">How to check your VA claim or appeal status online</h2><p>Sign in to VA.gov to check the status of your VA claim or appeal.</p>",
      "id": "b871d67e-44a1-45d7-a3c9-948795c0b10d",
      "type": "paragraph--wysiwyg",
    },
  ],
  "description": "Find out how to check the status of a VA claim or appeal online.",
  "entityId": 298,
  "entityPath": "/check-your-va-claim-or-appeal-status",
  "featuredContent": [
    {
      "html": "<h3>What you need to know</h3><p>You can check the status of your disability claim online.</p>",
      "id": "07b2ad1b-9b94-464a-8961-3ca73e84c8aa",
      "type": "paragraph--wysiwyg",
    },
  ],
  "id": "27aaaef6-0c67-4639-b7d7-84cb3c124cff",
  "introText": "<p>Find out how to check the status of a VA claim or appeal online.</p>
",
  "lastUpdated": "2020-04-30T20:11:16+00:00",
  "metatags": undefined,
  "moderationState": "published",
  "published": true,
  "relatedLinks": null,
  "showTableOfContents": true,
  "title": "Check Your VA Claim or Appeal Status",
  "type": "node--page",
}
`;
